CONTRIBUTING.md

```diff
@@ -1,181 +0,0 @@
-# Contribute To PyTorch/XLA
```

Comment:
IMO developer-oriented documentation (at least [`CONTRIBUTING`](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors#adding-a-contributing-file)) should stay at the top level or in `docs`, since contributors will probably look for instructions on our GitHub repository.

Moving API guide is a good idea, since that's user-facing.

Reply:
I moved the content to the docs/source directory because that is where Sphinx is looking for it. None of the docs outside of the source directory were being picked up by the doc build process. I'm OK with leaving CONTRIBUTING in the docs directory. I prefer to not have all docs in a single directory because that makes it difficult to find a file you want to update. We can add a comment in CONTRIBUTING about how we structure the docs to explain how to find a file.

Reply:
I will look into using markdown. The docs use the Pytorch Sphinx theme, perhaps there is a TOC setting that would help.

Reply:
I was able to get markdown files to work. I've pushed the updates to the branch I used when I created the PR.